File: Sorting Algos/cpp/bubble_sort.cpp
#include <iostream>
using namespace std;

void bubble_sort(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        bool swapped = false;
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(arr[j], arr[j + 1]);
                swapped = true;
            }
        }
        if (!swapped) break;
    }
}

void print_array(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main() {
    int n;
    cout << "Enter the number of elements: ";
    cin >> n;
    int* arr = new int[n];
    cout << "Enter the elements: ";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    bubble_sort(arr, n);
    cout << "Sorted array: ";
    print_array(arr, n);

    delete[] arr;
    return 0;
}

File: Sorting Algos/py/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True
        if not swapped:
            break
    return arr

while True:
    user_input = input("Enter numbers separated by spaces (or type 'exit' to quit): ")
    if user_input.lower() == 'exit':
        break
    try:
        arr = list(map(int, user_input.split()))
        sorted_arr = bubble_sort(arr)
        print("Sorted array:", sorted_arr)
    except ValueError:
        print("Please enter valid integers.")